app/models/show_series.rb

```diff
@@ -90,29 +93,39 @@ def slug_candidates
 
   def save_episodes
     if recurring?
-      start_day = DateTime.new self.start_date.year,
-                               self.start_date.month,
-                               self.start_date.day,
-                               self.start_time.hour,
-                               self.start_time.min
-      recurrences.each_with_index do |r|
+      first_recurrence = DateTime.new recurrences.first.year,
+                                      recurrences.first.month,
+                                      recurrences.first.day
+      start_day = DateTime.new first_recurrence.year,
+                               first_recurrence.month,
+                               first_recurrence.day,
+                               self.start_time.in_time_zone(self.time_zone).hour,
+                               self.start_time.in_time_zone(self.time_zone).min,
+                               self.start_time.in_time_zone(self.time_zone).sec,
+                               ActiveSupport::TimeZone[self.time_zone].formatted_offset
+
+      recurrences.each do |r|
         scheduled_show = self.episodes.new
         scheduled_show.radio = self.radio
         scheduled_show.dj = self.users.first # TODO drop dj_id from ScheduledShow?
-        scheduled_show.image = self.image if self.image.present?
-        new_date = DateTime.new r.year, r.month, r.day
-        difference_in_hours = (new_date - start_day).to_i * 24
-        start_at = start_day.advance(hours: difference_in_hours)
-        scheduled_show.start_at = start_at
-        scheduled_show.end_at = start_at + (self.end_time - self.start_time).seconds
-        scheduled_show.slug = nil
-        scheduled_show.title = self.title
-        if self.default_playlist.present?
-          scheduled_show.playlist = self.default_playlist
-        else
-          scheduled_show.playlist = self.radio.default_playlist
+        # scheduled_show.image = self.image if self.image.present?
+        new_start_at = DateTime.new(r.year, r.month, r.day, self.start_time.in_time_zone(self.time_zone).hour, self.start_time.in_time_zone(self.time_zone).min, self.start_time.sec, ActiveSupport::TimeZone[self.time_zone].formatted_offset)
+        if new_start_at > Time.now
+          difference_in_days = (new_start_at - start_day).to_i
+          # have to calculate time with .advance to preserve correct hour across time zone boundry
+          start_at = start_day.in_time_zone(self.time_zone).advance(days: difference_in_days)
```

Comment:
This will be off +1 hour during PDT ? 
due to calling `in_time_zone`

but the calculation is incorrect if we don't call `in_time_zone`.